src/app/fdctl/run/tiles/tiles.h

```diff
@@ -92,15 +92,28 @@ typedef struct {
   long  (*lazy                    )( fd_topo_tile_t * tile );
   ulong (*populate_allowed_seccomp)( void * scratch, ulong out_cnt, struct sock_filter * out );
   ulong (*populate_allowed_fds    )( void * scratch, ulong out_fds_sz, int * out_fds );
-  ulong (*loose_footprint         )( fd_topo_tile_t * tile );
   ulong (*scratch_align           )( void );
-  ulong (*scratch_footprint       )( fd_topo_tile_t * tile );
+  ulong (*scratch_footprint       )( fd_topo_tile_t const * tile );
   void  (*privileged_init         )( fd_topo_t * topo, fd_topo_tile_t * tile, void * scratch );
   void  (*unprivileged_init       )( fd_topo_t * topo, fd_topo_tile_t * tile, void * scratch );
 } fd_tile_config_t;
 
 FD_FN_CONST fd_tile_config_t *
-fd_topo_tile_to_config( fd_topo_tile_t * tile );
+fd_topo_tile_to_config( fd_topo_tile_t const * tile );
+
+static inline ulong
+fdctl_tile_align( fd_topo_tile_t const * tile ) {
+  fd_tile_config_t * config = fd_topo_tile_to_config( tile );
+  if( FD_LIKELY( config->scratch_align ) ) return config->scratch_align();
+  return 0UL;
```

Comment:
Default align of 0 seems like a bad idea. This should probably just error, but if you need a default 1, 8, 32 or something seem much better.